curve25519-dalek/src/backend/serial/u64/field.rs

```diff
@@ -234,32 +254,92 @@ impl<'a> Add<&'a FieldElement51> for &FieldElement51 {
             (original_limbs[3] + _rhs.limbs[3]) as u64,
             (original_limbs[4] + _rhs.limbs[4]) as u64,
         ]);
+        assume(field_element(&output) == (field_element(self) + field_element(_rhs)) % p());
         output
     }
 }
 
 impl<'a> SubAssign<&'a FieldElement51> for FieldElement51 {
-    fn sub_assign(&mut self, _rhs: &'a FieldElement51) {
+    fn sub_assign(
+        &mut self,
+        _rhs: &'a FieldElement51,
+    )
+    // VERIFICATION NOTE: PROOF BYPASS
+
+        requires
+            forall|i: int| 0 <= i < 5 ==> #[trigger] old(self).limbs[i] < (1u64 << 54),
+            forall|i: int| 0 <= i < 5 ==> #[trigger] _rhs.limbs[i] < (1u64 << 54),
+        ensures
+            forall|i: int| 0 <= i < 5 ==> #[trigger] self.limbs[i] < (1u64 << 52),
+            self.limbs == spec_reduce(
+                [
+                    ((old(self).limbs[0] + 36028797018963664u64) - _rhs.limbs[0]) as u64,
+                    ((old(self).limbs[1] + 36028797018963952u64) - _rhs.limbs[1]) as u64,
+                    ((old(self).limbs[2] + 36028797018963952u64) - _rhs.limbs[2]) as u64,
+                    ((old(self).limbs[3] + 36028797018963952u64) - _rhs.limbs[3]) as u64,
+                    ((old(self).limbs[4] + 36028797018963952u64) - _rhs.limbs[4]) as u64,
+                ],
+            ),
+            // VERIFICATION NOTE: tentative additional spec; is this correct?
+            field_element(self) == field_sub(field_element(old(self)), field_element(_rhs)),
+    {
         /* ORIGINAL CODE
         let result = (self as &FieldElement51) - _rhs;
         self.0 = result.0;
         */
         /* MODIFIED CODE */
-        proof {
-            assume(false);
-        }
         let result = &*self - _rhs;
-        proof {
-            assume(false);  // BECAUSE OF VERUS TRAIT ISSUES
-        }
         self.limbs = result.limbs;
+        assume(field_element(self) == field_sub(field_element(old(self)), field_element(_rhs)));
+        assume(forall|i: int| 0 <= i < 5 ==> self.limbs[i] < (1u64 << 52))
```

Comment:
I think this should be redundant, as all properties should follow from `result = sub(self, rhs)` and the spec of `Sub`.